### PR TITLE
Kapow!

### PR DESCRIPTION
**Pull request by jaredhirsch**
Node tests not 100% solid (test automation crew still cranking) but want to get this mainlined sooner than later.

### CODE REVIEW

**Comment by jaredhirsch** [comment]
odd that the merge isn't automagic. working on rebasing the branch now.

**Comment by jaredhirsch** [comment]
ok, that looks better. @lloyd, do your thang :-)

File: .gitignore [hunk 1]
```diff
@@ -16,11 +16,6 @@
...
 /resources/email_templates/email-test.html
-/automation-tests/bid_selenium
+/automation-tests/node_modules
 /automation-tests/results
-/automation-tests/credentials.yaml
-/automation-tests/sauce.yaml
-automation-tests/persona_server/results/*
-/automation-tests/chromedriver.log
-/.coverage_data
-/cover_html
```
Comment (jrgm) [inline .gitignore:26]: Those last two `.coverage_data` and `cover_html` were added to `dev` in PR-2272 and should not be removed. But if it's a hassle to rebase again, don't worry; they'll eventually get added back in some other PR.

**Comment by jaredhirsch** [comment]
hell yes. vows finally giving jenkins something to work with. couple of broken tests, but breakage now leads to well-formed XML. we'll get to fixing the tests next :-)

**Comment by lloyd** [comment]
ok, debating whether landing in dev _right now_ is the right thing to do.  The only pre-requisites I would consider hard are:
1. an easy way to run all tests all browsers
2. test stability (90% of the time all tests all browsers succeeds)
3. jenkins integration (daily?)

This is a subset of what is mentioned in the planning etherpad. 

once 1 is done, I'd love to contribute to 2 by using it.

all that said, is there a good reason to merge into dev before these 3 top level requirements are met?

/cc @shane-tomlinson @jrgm

**Comment by lloyd** [comment]
yeah, and if 3 got done and we were running tests hourly using jenkins, then as soon as we have 50 runs with < 5 failures, we know we've attained 2...

**Comment by shane-tomlinson** [comment]
@lloyd - I agree with your requirements with the following caveats.

For #1, we have to define which browsers make this list and with what frequency tests are run against all browsers. Is this on checkin/push to origin, on merge, etc? What happens for developers that first push to their own repos and then submit a PR, do these checkins automatically kick off the full test set? Currently we have no way of telling travis "run Selenium tests from this repo, but not from that repo over there."

For #3, Jenkins integration against which branch - dev, stage, prod or against any old branch?

**Comment by jaredhirsch** [comment]
So, the tests look green on Jenkins with all-browsers selected: https://ci.mozilla.org/job/temporary-test-job/236/console . I'll put that job on a short cron to get some stability numbers, and update this ticket tomorrow/late tonight with details. [update--I got the env var wrong and was only running against chrome-vista, doh. more info soon.]

@shane-tomlinson, to address some of your questions:

**Which browsers should we use?** The browsers we worried about with our selenium tests in the past were the ones in the sauce-browsers list, excluding the mobile platforms @jrgm added, but those don't track precisely to our official list of supported browsers/platforms (see [MDN](https://developer.mozilla.org/en-US/docs/Persona/Browser_compatibility)). I'd say, running that set of browsers (those in sauce-browsers excluding mobile) is a good starting point.

Code proposal: we generalize the current browser info file, and pass the file name optionally on the command line. Since sauce-platforms is just an exported object, we can have a "core" file that's required by an "extras" list, to hold stuff like mobile browsers that aren't ready for primetime yet. (I'll copy this idea into the test-automation-spec etherpad so it doesn't get lost). This is a nice-to-have.

**Which branches do we run against, and how often?** This is mentioned at a high level in the [automated testing wiki page on github](https://github.com/mozilla/browserid/wiki/Automated-Browser-Testing). 
- We'll run all tests/all browsers against stage and prod on a long cron (every 4 or 6 hours). 
- We'll run just the health-check tests, on just one stable browser/OS combo, against stage and prod every 5 minutes, as a monitoring stopgap. (This is a temporary monitor and I'll be improving our monitoring after this crew spins down.) 
- We'll run all tests/all browsers against dev on git push, as well as on a 4 or 6 hour cron. 

This is all already implemented in Jenkins on ci.mozilla.org; I just need to update the invocations to use our node tests instead of the old py tests.

**What about Travis testing PRs where the branch lives on somebody's fork?** I dunno. That's a good question. I guess Travis needs to give the git repo URL along with the SHA? Is that a tenable solution? Otherwise, maybe we can amend our git workflow practices to submit PRs from a branch living on mozilla/browserid.

**Comment by shane-tomlinson** [comment]
@6a68:

> What about Travis testing PRs where the branch lives on somebody's fork? I dunno. That's a good question. I guess Travis needs to give the git repo URL along with the SHA? Is that a tenable solution? Otherwise, maybe we can amend our git workflow practices to submit PRs from a branch living on mozilla/browserid.

This is what I suspected, so I did that yesterday. ;) What I am trying to get at is an ACL of some sort. What I have done as a super basic measure is to create a whitelist set of repo owners whose repo's can be tested via testy. A POST request to testy passes the git repo URL as well as the sha. Testy looks at the repo URL, divines the repo owner, checks against the whitelist and proceeds.

This is a start, but it is still open to abuse as any random yahoo could POST to testy the URL of one of the approved owner's repos and kick off sauce tests.

What I would like is to whitelist a set of IP addresses so that we can check the ip address of the requester, and whitelist Travis IP addresses only or use HTTP authentication with a secret stored on travis to sign in. Dunno. The former is not yet possible as the awsbox proxy does not forward the ip address of the requester from the proxy to the main server. The latter I thought the whole reason we were doing this exercise is so we didn't have to store secrets on Travis.

**Comment by jaredhirsch** [comment]
OK, so we've got 90%+ stability on all platforms except those added really recently (mac safari 5, win 8 IE10, mobile).

Suggest merging in, stabilizing those new platforms separately, and merging those fixes later.

Thoughts?

**Comment by lloyd** [comment]
dig it, will do a pass in the morning and merge!

**Comment by jaredhirsch** [comment]
zomg so many commits! looking stable on FF 16/Vista, what else needs to happen to merge this puppy in today?

**Comment by jaredhirsch** [comment]
r+ from lloyd in IRC. merging.

**Pull request merged**